### PR TITLE
final fix of bugs

### PR DESCRIPTION
**Pull request by Pragati1610**

### CODE REVIEW

File: Login/routes/login.js [hunk 1]
```diff
@@ -18,7 +19,11 @@ router.post("/signup", (req, res) => {
         return res.status(409).json({
           message: "mail exists",
         });
-      } else {
+      } else if (pass.length < 7){
```
Comment (L04DB4L4NC3R) [inline Login/routes/login.js:22]: DO this low computation check before querying DB

File: Login/routes/login.js [hunk 2]
```diff
@@ -108,23 +113,8 @@ router.post("/login", (req, res) => {
...
-        userid: user._id,
-      });
-    })
-    .catch((err) => {
-      console.log(err);
-      res.status(500).json({
-        error: err,
-      });
-    });
+router.post("/guest", (req, res)=>{
```
Comment (L04DB4L4NC3R) [inline Login/routes/login.js:116]: Remove this route

File: Login/routes/oAuth.js [hunk 3]
```diff
@@ -35,6 +35,10 @@ async function getGoogleDriveFiles(access_token) {
   return data;
 }
 
+router.post("/google", async (req, res) => {
```
Comment (L04DB4L4NC3R) [inline Login/routes/oAuth.js:38]: Why is this empty?

File: Login/routes/patch1.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+
+const router = express.Router();
+
+HEROKU_CRED = process.env.HEROKU_CRED
```
Comment (L04DB4L4NC3R) [inline Login/routes/patch1.js:5]: What does this contain, and if you are using it as a secret then why are you sending it out to the user in the `/auth2check` route.

File: app.js [hunk 5]
```diff
@@ -23,6 +23,7 @@ const { check, validationResult } = require("express-validator");
 const loginRoute = require("./Login/routes/login");
 const oAuthRoute = require("./Login/routes/oAuth");
 const previewRoute = require("./Login/routes/preview");
+const patch1Route = require('./patch3');
```
Comment (L04DB4L4NC3R) [inline app.js:26]: Need better name then `./patch3`

File: deploy.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+docker stack rm groff
+docker-compose pull
+docker stack deploy -c docker-compose.yml
```
Comment (L04DB4L4NC3R) [inline deploy.sh:3]: No need to do these 3 commands. You can use the `--force` command to update the docker stack with the latest images.

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+version: '3.7'
+services:
+  api:
+    container_name: backend
```
Comment (L04DB4L4NC3R) [inline docker-compose.yml:4]: Dont use container_name in stack based deployment

**Review by L04DB4L4NC3R — CHANGES_REQUESTED** [review]
Change the name of the PR also.

File: Login/routes/patch1.js (hunk 4 above)
Comment (1UC1F3R616) [reply Login/routes/patch1.js:5]: ok, will fix this

File: docker-compose.yml (hunk 7 above)
Comment (Pragati1610) [reply docker-compose.yml:4]: Noted, Have made the change

File: deploy.sh (hunk 6 above)
Comment (Pragati1610) [reply deploy.sh:3]: Noted, Have made the change

File: app.js (hunk 5 above)
Comment (Pragati1610) [reply app.js:26]: Noted, Have made the change

File: Login/routes/oAuth.js (hunk 3 above)
Comment (Pragati1610) [reply Login/routes/oAuth.js:38]: Noted, Have deleted the route, was unused

File: Login/routes/login.js (hunk 2 above)
Comment (Pragati1610) [reply Login/routes/login.js:116]: Noted, Have made the change

**Pull request merged**